src/components/Wallet/AccountSelector/AccountSelector.tsx

```diff
@@ -0,0 +1,119 @@
+import { MutableRefObject, useMemo } from 'react';
+import { Account } from '../../../generated/graphql';
+import { AccountItem } from './AccountItem/AccountItem';
+import { Button, ButtonKind } from '../../Button/Button';
+import './AccountSelector.scss';
+
+export interface AccountSelectorProps {
+  accounts?: Account[];
+  account?: Account;
+  onAccountSelected: (asset: Account) => void;
+  innerRef: MutableRefObject<HTMLDivElement | null>;
+  closeModal: () => void;
+  setActiveAccount: Function;
+  isExtensionAvailable: boolean;
+}
+
+/**
+ * Renders a list of accounts that the user can select an account from
+ */
+export const AccountSelector = ({
+  accounts,
+  onAccountSelected,
+  account,
+  innerRef,
+  closeModal,
+  setActiveAccount,
+  isExtensionAvailable,
+}: AccountSelectorProps) => {
+  const activeAccount = useMemo(() => account, [account]);
+
+  return (
+    <div className="account-selector" ref={innerRef}>
+      <div className="account-selector__content-wrapper">
+        <div className="d-flex flex-align-space mx-3 my-3 account-selector__heading">
+          <div>
+            {isExtensionAvailable ? 'Select an account' : 'Install extension'}
+          </div>
+          <div
+            className="account-selector__close-modal-btn"
+            onClick={() => closeModal()}
+          >
+            x
+          </div>
+        </div>
+        {isExtensionAvailable ? (
+          <>
+            {accounts?.length ? (
+              <div className="account-selector__accounts-list">
+                {accounts?.map((account, i) => (
+                  <AccountItem
+                    key={i}
+                    onClick={() => onAccountSelected(account)}
+                    active={account.id === activeAccount?.id}
+                    account={account}
+                    setActiveAccount={setActiveAccount}
+                  />
+                ))}
+              </div>
+            ) : (
+              <div className="mx-3 my-5 text-center">
+                <h4>No accounts available</h4>
+                <a href="/#" className="account-selector__create-account-link">
+                  Need help creating an account? <br />
+                  Click here
+                </a>
+              </div>
+            )}
+            {account && (
+              <div className="d-flex mx-3">
+                <Button
+                  kind={ButtonKind.Secondary}
+                  onClick={() =>
+                    setActiveAccount({
```

Comment:
Waits for the mutation to finish, then executes the handler/callback? I think we should just fire the `onAccountSelected` and deal with the `setActiveAccount` on a container level as i've mentioned earlier in this review.

Then in the container itself, you can run the mutation and THEN close the modal. It could be done in the following way:

```
type onAccountSelected = (account: Account, closeModel: () => void) => void
```

And then once you've handled `onAccountSelected`, you can opt in to execute the component's internal `closeModal` from the container.

I am not entirely sure about this approach, but just dropped my thoughts here.

Reply:
Fixed along with refactoring